tsdb/engine/tsm1/file_store.go

```diff
@@ -1356,12 +1362,12 @@ func DefaultParseFileName(name string) (int, int, error) {
 
 	generation, err := strconv.ParseUint(id[:idx], 10, 32)
 	if err != nil {
-		return 0, 0, fmt.Errorf("file %s is named incorrectly", name)
+		return 0, 0, fmt.Errorf("file %s is named incorrectly: %w", name, err)
```

Comment:
I would add that we were attempting to parse the generation number to the error context.